### ISSUE TITLE
Missing weekly data

### ISSUE THREAD

**Issue by JoeSydlowski** [jchernak96/NFL-Injury-Data-PFR-#1]
I noticed that Julio Jones doesn't have any weekly data for 2017. Is that something you can fix in the scrape or is it a PFR issue?

```
library(tidyverse)
read_csv("https://raw.githubusercontent.com/jchernak96/NFL-Injury-Data-PFR-/master/Data/PFR_2017_Injuries.csv") %>% 
  filter(Name == "Julio Jones")
```

**Comment by JoeSydlowski** [comment]
Actually here are the top 30 fantasy relevant seasons with NULLs in the weekly snap data in case the problem extends past Julio

![image](https://user-images.githubusercontent.com/47405249/116029896-f0dacc00-a61f-11eb-9a57-54d84cf8fdcc.png)

**Comment by jchernak96** [comment]
Good catch, Julio is weird because he didn't have any injury according to PFR in 2017. He is in the data set but under the name abbreviation of "Ju.Jones" for 2017. This is because of the following process I followed when constructing the data

1. Scraped data from PFR. If a player did not experience any injury that season they will not show up in the data. So for example: Julio Jones in 2017 did not experience an injury and thus is not even on the PFR injury page.
2. Added in PFO snap count data and merged by name abbreviation, team, and season. Since some players did not experience an injury but played, this merge adds data on players who played the season fully healthy which we didn't have from PFR. 
3. Since Julio didn't get hurt, the PFO didn't could not find him and just added a new row of data for him. Now the weird part is for 2017 they used the name abbreviation of "Ju.Jones" rather than "J.Jones" which is why when I also merged in mrcaseb data of positions and player info, he wasn't matched for that season. 

Let me know if you spot any other issues or if that doesn't make sense!

**Comment by JoeSydlowski** [comment]
Yeah, that makes sense! Looks like the same thing is happening on PFO with Metcalf where he's listed as "DK. Metcalf" in 2019 and "D. Metcalf" in 2020. 
I've never scraped PFO before so I'm not sure what the process is there, but if you're catching weekly snap data from PFO for other fully healthy players (that aren't coming from PFR) why would it exclude Julio's weekly snap data. If the problem is coming from the join to mrcaseb's roster data would it be hard to code in a case when to catch a few of these odd cases?

Appreciate all the work you've done to put this data together! Very easy to join in to nflfastr data.

**Comment by jchernak96** [comment]
Glad you like the data and are putting it to good use!! I appreciate you pointing out the issues. 

I think the issue you mean is the random one row of Julio Jones where it just has a NA in week but his full personal details? When I merged in mrcaseb's roster data I set a condition in the merge to keep all data (so if the individual wasn't found it wouldn't remove anybody who was not found if that makes sense). So that means some players are just rows from his roster data cause they were not matched to anybody. Those players should have an "NA" in the week so that shouldn't be too tough to filter out. It might be a bit difficult to catch and fix those fringe cases just cause it is an abbreviated name and some players have pretty general names so it might not be perfect. I am open to suggestions on those though!

**Comment by JoeSydlowski** [comment]
Okay I'm getting a better understanding. mrcaseb has Julio's 2017, PFR doesn't have any injury data for Julio 2017, and PFO has snap data for him, but under Ju.Jones so it doesn't join, right? I think what I would do is use a case when to change a few of the names in the PFO data so that they do match to mrcaseb's roster data.
e.g.
`case_when(name == "Ju.Jones" & team == "ATL" ~ "J.Jones", name == "DK. Metcalf" ~ "D. Metcalf", TRUE ~ name)`
I understand if that's not something you want to maintain though, but luckily it doesn't happen very often so I think this is still the best snap data out there regardless.

**Comment by jchernak96** [comment]
That is correct, sorry for the delayed response! Thanks for the suggestion, I've done a lot of data cleaning with this data so I think the next step would to try out that "case when" argument and see how it does.
